weave-js/src/components/FancyPage/useProjectSidebar.ts

```diff
@@ -14,6 +15,18 @@ export const useProjectSidebar = (
   isLaunchActive: boolean = false,
   isWandbAdmin: boolean = false
 ): FancyPageSidebarItem[] => {
+  // must preserve named workspace url query string
+  // when navigating between workspace & runs pages
+  let nwSearchStr: string | undefined;
+  const location = useLocation();
+  const allSearchParams = new URLSearchParams(location.search);
+  const nwId = allSearchParams.get('nw');
+  if (nwId) {
+    const nwSearchParams = new URLSearchParams();
+    nwSearchParams.append('nw', nwId);
+    nwSearchStr = nwSearchParams.toString();
+  }
```

Comment:
specifically picks out only `nw` param and not other things like panel display names, etc